frontend/src/components/resourceMap/graph/graphFiltering.ts

```diff
@@ -0,0 +1,100 @@
+import { getStatus } from '../nodes/KubeObjectStatus';
+import { GraphEdge, GraphNode } from './graphModel';
+
+export type GraphFilter =
+  | {
+      type: 'name';
+      query: string;
+    }
+  | {
+      type: 'hasErrors';
+    }
+  | {
+      type: 'namespace';
+      namespaces: Set<string>;
+    }
+  | {
+      type: 'related';
+      id: string;
+    }
+  | {
+      type: 'custom';
+      label: string;
+      filterFn: (node: GraphNode) => boolean;
+    };
+
+/**
+ * Filters the graph nodes and edges based on the provided filters
```

Comment:
Can you please add in the docs if it does AND or OR? From the code it seems it matches any of the filters, not all the filters.

Something like?
> * The filters are applied using an OR logic, meaning that a node will be included if it matches any of the filters.